source/test/TestDirector.cc

```diff
@@ -145,6 +145,22 @@ bool TestDirector::test(const TestData &data) {
         return false;
     }
 
+    if (m_versionMinor >= ADDED_EXT_VEL) {
+        const auto &extVel = object->extVel();
+        if (data.extVel != extVel) {
+            logVectorDesync(data.extVel, extVel, "extVel");
+            return false;
+        }
+    }
+
+    if (m_versionMinor >= ADDED_INT_VEL) {
+        const auto &intVel = object->intVel();
+        if (data.intVel != intVel) {
+            logVectorDesync(data.intVel, intVel, "intVel");
+            return false;
+        }
+    }
```

Comment:
Did the `switch` fall-through approach not work well? In my head I still envision that as being a much cleaner solution

```
switch(m_versionMinor) {

case ADDED_INT_VEL:
    {
        const auto &intVel = object->intVel();
        if (data.intVel != intVel) {
            logVectorDesync(data.intVel, intVel, "intVel");
            return false;
        }
    }
case ADDED_EXT_VEL:
    {
        const auto &extVel = object->extVel();
        if (data.extVel != extVel) {
            logVectorDesync(data.extVel, extVel, "extVel");
            return false;
        }
    }
default:
    // The other checks
}
```